Initial converted configuration documentation for v0.1 (#4)
CDCgov/prime-fhir-converter: docs/fhir-hl7v2-converter/configuration.md

that can be reused, but that the result will be dependent on what constants and focus resource is set on the element using it.
4. Constants used as literals in FHIR Path expressions need to be surrounded by quotes (e.g. `myConstant: '"my literal string"'`)

Constants are used as follows:
* **For FHIR Path expressions** please refer to the
[FHIR Path specification section on environment variables](http://hl7.org/fhirpath/N1/#environment-variables).
Constants must have proper FHIR Path expression syntax including for string literals. This library also supports
having constants as a prefix by using the syntax `` %`<constant_name>-<prefix> ``
```yaml
  constants:
    diagnostic: 'Bundle.entry.resource.ofType(DiagnosticReport)[%orderIndex]'
    rsext: '"https://reportstream.cdc.gov/fhir/StructureDefinition/"'
  resource: '%resource.extension(%`rsext-parent-observation-value-descriptor`)'
```

* **For HL7 location spec**, all constants are treated as string replacements. The special syntax of
`%{<constant_name>}` is used 
```yaml
  constants:
     fieldPath: '/PATIENT_RESULT/ORDER_OBSERVATION(%{orderIndex})/OBR'
  hl7Spec: [ '%{fieldPath}(%{entityIdIndex})-3' ]
```

# Extending Schemas

> Top level schemas can extend and existing top level schema in order to reduce duplicate elements and facilitate

Reviewer: ```suggestion
Top level schemas can extend any existing top level schema in order to reduce duplicate elements and facilitate
```

customization for different purposes. You can chain as many top level schemas as you wish. Note that reusing an element

> name will result in overwriting the existing element. 

Reviewer: This is kind of where the confusion might set in, but I think it's helpful to elaborate on what elements can be overwritten.
```suggestion
name will result in overwriting the existing element, whether the existing element is at the top level of the extended schema or one of its child schemas. 
```

```yaml
hl7Type: ORU_R01
hl7Version: 2.5.1
extends: ../ORU_R01/ORU_R01-base
elements:
  # Disable an element. Note we only need the name and what is overwritten.
  - name: filler-order
    condition: 'false'
```

Extended schemas can:
* Define a different type and version of Hl7 v2 message than the top level schema

> * Override one or more elements based on the element name

Reviewer: We go back and forth between `overwrite` and `override` a bit, we might want to go through and make it consistent throughout, though I don't have a strong opinion of which of the two to use.

Author: Yeah. I did that change and did not go through to make sure everything is the same. Will update

>   * You override element can reside at the top level element list regardless where they reside in the inherited schema

Reviewer: ```suggestion
  * The overriding element can reside at the top level element list regardless where they reside in the inherited schema
```
